File: php/kaltura-api-notifications-handler-php/sync.php
<?php 

require_once('./session_config.php');
require_once('./script_config.php');
require_once('./lib/autoload.php');

// Create console in order to log activity
$console = new \Kaltura\Output\Console();
$console->startLog();

$params = $_POST;

try {
	if (!count($params)) {
		$console->log('No notification params');
		$console->log('');
		$console->log('=============');
		$console->log('  ALL DONE!');
		$console->endLog('=============');
	} else {
		// Instanciate a nofication processor object
		// First param is the notification parameters
		// Second param is validateSignature of the notification (make sure it comes from Kaltura)
		// Third param is the Kaltura admin secret

		// This will set up the notification client and check the notification signature
		$notificationProcessor = new \Kaltura\Notification\Processor($params, false, KALTURA_ADMIN_SECRET);

		// Setting up handler for 'entry_update' notification type
		$synchEntryHandler = new \Kaltura\Notification\Handler\SyncEntry(\Kaltura\Notification\Type::NOTIFICATION_TYPE_ENTRY_UPDATE);
		// Passing console to the handler so it can log as well
		$synchEntryHandler->setConsole($console);
		// Passing extra parameters from the config to the handler
		$synchEntryHandler->addData($handlerParams);

		// Adding handler to the notification processor
		// Note that you can set up multiple handlers, the processor will execute them in order
		$notificationProcessor->addHandler($synchEntryHandler);

		// Processing notification handler(s)
		$notificationProcessor->execute();

		$console->log('');
		$console->log('=============');
		$console->log('  ALL DONE!');
		$console->endLog('=============');
	}
} catch (Exception $e) {
	$console->log('');
	$console->timeLog("\n");
	$console->log('  /!\\ An error occurred!');
	$console->log('  '.$e->getCode().': '.$e->getMessage());
	$console->log('  '.$e->getTraceAsString());
	$console->log('');
	$console->log('  ======================================');
	$console->log('  END WITH ERRORS');
	$console->endLog('  ======================================');
}

?>

File: java/NotificationHandler/src/lib/Kaltura/Sync.java
package lib.Kaltura;

import java.io.PrintWriter;
import java.io.StringWriter;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;

import lib.Kaltura.Notification.Notification;
import lib.Kaltura.Notification.Processor;
import lib.Kaltura.Notification.Handler.NotificationSampleHandler;
import lib.Kaltura.Notification.Types.HandleEventType;
import lib.Kaltura.Notification.Types.NotificationType;
import lib.Kaltura.Output.Console;
import lib.Kaltura.Output.OutputInterface;
import lib.Kaltura.Output.StandaradOutput;
import lib.Kaltura.config.SessionConfig;

public class Sync {
	
	public static void sync(Map<String, String> params) throws Exception {
		
		// Create console in order to log activity
		List<OutputInterface> output = new ArrayList<OutputInterface>();
		output.add(new StandaradOutput());
		Console console = new Console(output);
		console.start();
		
		try {
			if (params.isEmpty()) {
				console.write("No notification params");
			} else {
				
				// Accept a single notification
				Notification notification = new Notification(params, true, SessionConfig.KALTURA_ADMIN_SECRET);
				
				// Create handler
				Set<HandleEventType> types = new HashSet<HandleEventType>();
				types.add(NotificationType.ENTRY_UPDATE);
				NotificationSampleHandler sampleHandler = new NotificationSampleHandler(types);
				sampleHandler.setConsole(console);
				
				// Create processor and assign the handler to it
				Processor notificationProcessor = new Processor(console);
				notificationProcessor.addHandler(sampleHandler);
		
				// Handle single notification
				notificationProcessor.execute(notification);
			}
			
			console.write("");
			console.write("=============");
			console.write("  ALL DONE!");
			console.write("=============");
			console.end();
			
		} catch (Exception e) {
			StringWriter sw = new StringWriter();
			e.printStackTrace(new PrintWriter(sw));
			String exceptionAsString = sw.toString();
			
			console.write("");
			console.write("  An error occurred!");
			console.write("  " + e.getCause() + ": " + e.getMessage());
			console.write("  " + exceptionAsString);
			console.write("");
			console.write("  ======================================");
			console.write("  END WITH ERRORS");
			console.write("  ======================================");
			console.end();
		}
	}
	
}